fitsnap3/fitsnap.py

```diff
@@ -80,4 +80,7 @@ def perform_fit(self):
     def write_output(self):
         if not config.args.perform_fit:
             return
-        output.output(self.solver.fit, self.solver.errors)
+        # Prevent Output from trying to process non-existing solver.fit when using PyTorch.
+        if (config.sections["SOLVER"].solver != "PYTORCH"):
```

Comment:
Both JAX and PYTORCH should not output generic output, and probably not any nonlinear solver.
My suggestion is to use the linear member of the solver class.
```
if self.solver.linear:
    output.output(self.solver.fit, self.solver.errors)
```